src/engraving/dom/guitarbend.cpp

```diff
@@ -484,6 +484,7 @@ GuitarBendSegment::GuitarBendSegment(GuitarBend* sp, System* parent)
     : LineSegment(ElementType::GUITAR_BEND_SEGMENT, sp, parent, ElementFlag::MOVABLE)
 {
     m_text = new GuitarBendText(this);
+    m_text->setParent(this);

```

Comment:
Don't we need it in the copy constructor too? Just to be safe, and for consistency perhaps. 
(Also I think there is a setParent somewhere in the layout code too, so that one might become redundant)

Reply:
> Don't we need it in the copy constructor too?

Ah yes, thanks. I've honestly never understood why the top-most constructor (i.e. the EngravingObject) takes in the parent items as a constructor parameter but _doesn't_ set the explicit parent with it. 

> (Also I think there is a setParent somewhere in the layout code too, so that one might become redundant)

It may well be, but I'd rather not make that change now for the patch.

Reply:
> Ah yes, thanks. I've honestly never understood why the top-most constructor (i.e. the EngravingObject) takes in the parent items as a constructor parameter but doesn't set the explicit parent with it.

This "explicit parent" business has some history; it actually stems from a never-finished refactor by Igor. I think it was this PR: https://github.com/musescore/MuseScore/pull/8996 (note that EngravingItem was back then still called Element (I miss that name), and EngravingObject was called ScoreElement (I don't miss that one)).

Before, Elements had a parent Element, and ScoreElement had a reference to the Score, but had no parent. The parent Element was set by calling `setParent`; not via the constructor.

I don't know what was the direct reason for that PR, but I think it's because there was no generality in when elements where children of each other or when their parent was nullptr, and that this caused problems with memory management, specifically during destruction. I guess it caused either leaks or crashes or both.

So the model was changed; now EngravingObject has a parent EngravingObject, and the parent is passed via the constructor. Instead of setting the parent to nullptr, the only option is now to set the parent to some dummy item.

However, a lot of existing code checked whether some elements had parent `nullptr`. But now, parent is never nullptr, so the behaviour of that code would change. To avoid that, the `m_isParentExplicitlySet` member was introduced. This member is basically used to simulate the old parent behaviour: until you call `setParent`, it is pretended that there is no parent, even though there is one (namely the one you passed to the constructor).

The ultimate goal is to get rid of `m_isParentExplicitlySet`, and I believe/hope also of `DummyItem`. However, to achieve that, we need to check all cases where `explicitParent` or `parentItem` is used, check whether they actually depend on whether the parent is set _explicitly_ or that it does not matter, and if it does matter, eliminate the fact that it matters. But nobody has ever worked on this task, and it is not documented anywhere. Actually, I should perhaps open a tech debt issue about that. 

> It may well be, but I'd rather not make that change now for the patch.

I agree, however let's make sure it happens eventually (perhaps as part of the above mentioned investigation task)